storage/storage_image.go

```diff
@@ -409,7 +409,9 @@ func (s *storageImageDestination) HasBlob(ctx context.Context, blobinfo types.Bl
 	if err != nil && errors.Cause(err) != storage.ErrLayerUnknown {
 		return false, -1, errors.Wrapf(err, `Error looking for compressed layers with digest %q`, blobinfo.Digest)
 	}
-	if len(layers) > 0 {
+	// If we got a match, check the compressed digest: if it is not equal, we have a layer which matches the searched
+	// one only after decompression. So, it is not the same layer.
```

Comment:
```suggestion
	// one only after decompression. (This can happen only due to a bug in earlier c/storage versions.)
	// So, the CompressedSize value may not match blob info.Digest, and we can’t use it.
```